tools/doc-gen/index.ts

```diff
@@ -0,0 +1,377 @@
+import { join } from 'path';
+import {
+  readFileSync,
+  readdirSync,
+  writeFileSync,
+  mkdirSync,
+  existsSync,
+} from 'fs';
+import ts from 'typescript';
+import type { TSESLint } from '@typescript-eslint/utils';
+
+// TODO: Add logic for cleaning output directory
```

Comment:
maybe rimfar or fs-extra. I'm sure there's a trillion ways to easily do this through node

Reply:
We've also gotta decide _when_ to nuke the docs directory-- before every run, or as a separate task? If we have multiple sub-folders, might just want to nuke the one we're targeting